documentation/src/docs/reader-group-design.md

```diff
@@ -9,85 +9,90 @@ You may obtain a copy of the License at
 
 Given this information:
 
-- A new reader can infer which segments are available to read from. (By virtue of it being absent from the state)
-- Dealing with a segment being merged becomes easy, because the last reader to reach the end of its pre-merge segment knows it can freely take ownership of the new segment.
-- Readers can see their relative load and how they are progressing relative to the other readers in their group and can decide to transfer segments if things are out of balance.
-- This allows readers to take action directly to ensure all the events are read without the need for some external tracker.
+ - A new Reader can infer which segments are available to read from. (By virtue of it being absent from the state).
+ - Merging segments becomes easy, because the last Reader to reach the end of its pre-merge segment knows it can freely take ownership of the new segment.
+ - Readers can see their relative load and how they are progressing relative to the other Readers in their group and can decide to transfer segments if things are out of balance.
+ - This allows Readers to take action directly to ensure all the events are read without the need for some external tracker.
+
+## Reader Group APIs
+
+The external APIs to manage Reader Groups could be added to the `StreamManager` object. It consist of:
 
-## ReaderGroup APIs
-The external APIs to manage ReaderGroups could be added to the StreamManager object. They consist of:
 ```
     ReaderGroup createReaderGroup(String name, Stream stream, ReaderGroupConfig config);
     ReaderGroup getReaderGroup(String name, Stream stream);
     void deleteReaderGroup(ReaderGroup group);
 ```
-When a ReaderGroup is created, it creates a [StateSynchronizer](state-synchronizer-design.md) shared by the readers. To join a ReaderGroup readers would just specify it in their configuration:
+When a Reader Group is created, it creates a [State Synchronizer](state-synchronizer-design.md) shared by the Readers. To join a Reader Group, Readers would just specify it in their configuration:
+
 ```
-    ReaderConfig cc = new ReaderConfig(props);
-    Reader<T> reader = a_stream.createReader("my_reader_id", "my_reader_group", cc);
+ClientFactory clientFactory = ClientFactory.withScope(scope, controllerURI);
+EventStreamReader< T > reader = clientFactory.createReader(readerId, READER_GROUP_NAME, serializer, readerConfig);
+
 ```
-When readers join the group they use the state to determine which segments to read from. When they shut down they update the state so that other readers can take over their segments.
+The Readers, while joining the group access the information stored on the state to determine which segments to read from. Once when they shut down, they update the state so that other Readers can take over their segments.
 
 # Failure detector
```

Comment:
I don't understand the structure of the documentation at this point. We start talking about `failure detector` but the associated text is not explaining it correctly. Even worse, a similar piece of content is repeated later on in `Reader going offline`. 

I suggest to refactor this part as follows:
1.- Replace `Failure detector` by something like `Reader Group Lifecycle: Example Operations`. Then, we can introduce the section with something like: `In what follows, we describe the workflow of the main operations executed during the liefecycle of a Reader Group.`
2.- Delete the content until `New Reader`.
3.- Then, each subsection subsection describes a different event within a Reader Group: `New Reader`, `Segments Merging`, `Reader Going Offline`. 

This structure may be extended to describe other Reader Group operations in the future.

Reply:
@RaulGracia SO shoul d i remove the para starting with `We still need some effective mechanism`................till `exactly once processing guarantees.`